### ISSUE TITLE
Studio may not refresh or refresh with errors

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/density#180]
**Device**
Mac Book Pro
**OS**
MacOS 13.4
**Browser**
Firefox
**Problem Description**
For https://github.com/phetsims/qa/issues/942.
When refreshing the page for Studio in Firefox on Mac, the page sometimes either fails to load, or loads with errors as shown in the screenshot. I'm not sure how to reproduce every time, but it seems to happen every other refresh attempt.
**Visuals**
<img width="1286" alt="Screenshot 2023-05-19 at 2 21 50 PM" src="https://github.com/phetsims/density/assets/41024075/c9685518-1976-409a-9997-e83079935a5c">

**Comment by jonathanolson** [comment]
@samreid or @zepumph, any idea what this is? I'm currently unable to reproduce.

**Comment by zepumph** [comment]
Interesting bug. I am wondering if you can try to reproduce this on master? I updated the assertion to try to give us a bit more data about why this is happening on refresh.

**Comment by zepumph** [comment]
@samreid and I investigated a bit further, and couldn't reproduce. We are leaning towards thinking of this as non-blocking. The above type error is so wild it is confusing us how it could ever get triggered. 

We haven't been changing any of the code around this failure. 

Next steps are for @KatieWoe and us to pair for reproducing, and to have this no longer block the density publication.

**Comment by samreid** [comment]
@zepumph and I worked with @KatieWoe to reproduce the problem.  We feel it may be related to https://github.com/phetsims/phet-io-wrappers/issues/342#issuecomment-638472518, so we will commit a proposed fix and see if it fixes the problem on the Mac/Firefox.  NOTE: I still can't reproduce this studio problem on my side.

**Comment by zepumph** [comment]
@KatieWoe, @samreid and I were able to great a reproducible case on SR's mac + firefox following steps in https://github.com/phetsims/phet-io-wrappers/issues/342#issuecomment-638472518, and would like you to see if the above patch fixes things by testing out with this new dev version.

Please also be on the lookout for the more crazy bug that has an error from `selectScreenTreeNodeFromScreenChange`. I'm unsure if this will have fixed that one necessarily. 

https://phet-dev.colorado.edu/html/density/1.2.0-dev.1/phet-io/

**Comment by KatieWoe** [comment]
I encountered the bug right away again in the above test. So far only the won't load bug. I can keep trying for the more crazy bug if you want. This was again seen with dev tools closed.

**Comment by samreid** [comment]
Can you clear the firefox caches and check if 1.2.0-dev.1 or the other one remain buggy?

**Comment by samreid** [comment]
When @KatieWoe cleared the cache each time before the test, the problem went away.

**Comment by pixelzoom** [comment]
Is this problem present in https://github.com/phetsims/qa/issues/947?  If so, can we get an issue geometric-optics?

**Comment by arouinfar** [comment]
I don't think this issue should block Density publication. @zepumph expressed a similar sentiment in https://github.com/phetsims/density/issues/180#issuecomment-1563521764, so I feel like we can proceed with the Density RC spot-check (all other issues have already been addressed and cherry-picked).

**Comment by KatieWoe** [comment]
I did see this in Geometric Optics. Linking the test here https://github.com/phetsims/qa/issues/947

**Comment by pixelzoom** [comment]
@arouinfar said:
> I don't think this issue should block Density publication. ...

This problem crashes Studio when refreshing the page, and QA reports that it happens often.  Presumably clients will also be refreshing the page.  So why is this not a blocking issue for a PhET-iO release?

**Comment by pixelzoom** [comment]
Based on https://github.com/phetsims/density/issues/180#issuecomment-1570595332 ...

My understanding is this was reviewed, and the fix didn't work.  So removing the "ready-for-review" label.

**Comment by arouinfar** [comment]
> This problem crashes Studio when refreshing the page, and QA reports that it happens often. Presumably clients will also be refreshing the page. So why is this not a blocking issue for a PhET-iO release?

From my understanding, Studio will sometimes load with an error or needs to be refreshed an extra time. @KatieWoe said clearing the cache fixes the issue. This has been difficult to reproduce. I cannot reproduce. @zepumph and @samreid were unable to reproduce initially and had to pair with @KatieWoe to figure out the reproducible case. The reproducible case they linked back to was from nearly 3 years ago (https://github.com/phetsims/phet-io-wrappers/issues/342#issuecomment-638472518). If this bug has been around for that long, presumably it's in published simulations. It doesn't make sense to me to completely block PhET-iO publication for a bug like this.

Also, it's not just me saying it's not publication-blocking. @zepumph said the same thing in https://github.com/phetsims/density/issues/180#issuecomment-1563521764.

**Comment by zepumph** [comment]
Discussed during PhET-iO meeting today. 

This does not block publication of Density or GO. 

**If there is a bug in all of our published PhET-iO sims, it does not need to block publication of another simulation.** Sometimes it is still easier to just fix it eagerly for the new publication. 

Should we apply a fix for this to all published PhET-iO sims? 
SR: We need to continue to investigate at a high priority. Then once we know all info, yes. This is an important bug that should be fixed, and could likely be effecting our client wrappers now..

BM: We should find out how many firefox users are using PhET-iO sims. Globally its only 5% of users. 

@samreid and I will continue to investigate. I wonder if there is a way to fix our phetioInitialized message to get around this problem.

**Comment by samreid** [comment]
Since this is non-blocking for the sim, we would like to move it to the studio/ repo.
